hal_silabs: Compiling RAIL for efr32xg1x series with rail.h fails

**Describe the bug**
Compiling a file that includes `rail.h` for the efr32xg1x series fails the build.
This happens because one of the files copied from the gecko SDK was copied from an incorrect version.
Fixed by: https://github.com/zephyrproject-rtos/hal_silabs/pull/45 and https://github.com/zephyrproject-rtos/zephyr/pull/63476

**To Reproduce**
Steps to reproduce the behavior:
1. Add `#include <rail.h>` to any source file (e.g. `samples/hello_world/src/main.c`)
2. Build `west build -b efr32_radio_brd4255a samples/hello_world`
3. See error

**Expected behavior**
Build successfully

**Impact**
Can't use RAIL on 1x devices.

**Logs and console output**
```
In file included from /home/user/zephyrproject/zephyr/samples/hello_world/src/main.c:8:
/home/user/zephyrproject/modules/hal/silabs/gecko/platform/radio/rail_lib/common/rail.h:2654:7: error: unknown type name 'RAIL_PaPowerSetting_t'
 2654 | const RAIL_PaPowerSetting_t *RAIL_GetPowerSettingTable(RAIL_Handle_t railHandle, RAIL_TxPowerMode_t mode,
      |       ^~~~~~~~~~~~~~~~~~~~~
/home/user/zephyrproject/modules/hal/silabs/gecko/platform/radio/rail_lib/common/rail.h:2671:38: error: unknown type name 'RAIL_PaPowerSetting_t'; did you mean 'RAIL_TxPowerConfig_t'?
 2671 |                                      RAIL_PaPowerSetting_t paPowerSetting,
      |                                      ^~~~~~~~~~~~~~~~~~~~~
      |                                      RAIL_TxPowerConfig_t
/home/user/zephyrproject/modules/hal/silabs/gecko/platform/radio/rail_lib/common/rail.h:2683:1: error: unknown type name 'RAIL_PaPowerSetting_t'; did you mean 'RAIL_TxPowerConfig_t'?
 2683 | RAIL_PaPowerSetting_t RAIL_GetPaPowerSetting(RAIL_Handle_t railHandle);
      | ^~~~~~~~~~~~~~~~~~~~~
      | RAIL_TxPowerConfig_t
[18/123] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /snap/bin/cmake --build /home/user/zephyrproject/zephyr/build
```

--- a/west.yml
+++ b/west.yml
@@ -222,7 +222,7 @@ manifest:
       groups:
         - hal
     - name: hal_silabs
-      revision: 6a4521e5934d2e787f782b9e5370f3206e6cc9d2
+      revision: d191d981c4eb20c0c7445a4061fcdbcfa686113a
       path: modules/hal/silabs
       groups:
         - hal

